### PR TITLE
[PSDK-360] Support gasless sends

### PR DESCRIPTION
**Pull request by alex-stone**
### What changed? Why?
This adds support for gasless transfers initially for `USDC` on Base.

This can be used using the optional `gasless` keyword argument to the `transfer` function.

```
transfer = address.transfer(15, :usdc, '0x1234', gasless: true)
```

### Example Sponsored USDC Send
https://basescan.org/tx/0x4e8a04014365597474158460f9e2884c8d05b21ce1f6c08b87e2520b9ddd0a64

### Gas Sponsorship Mechanism:

This utilizes https://eips.ethereum.org/EIPS/eip-3009 and signes the `TransferWithAuthorization` EIP-712 message.

#### Qualified Impact

This adds a new optional feature and does not break existing usage

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -212,6 +212,19 @@ t = w1.transfer(0.00001, :eth, w2).wait!
 puts "Transfer successfully completed: #{t}"
 ```
 
+### Gasless USDC Transfers
+
+To transfer USDC without needing to hold ETH for gas, you can use the `transfer` method with the `gasless` option set to `true`.
```
Comment (jazz-cb) [inline README.md:217]: what is our intuition behind setting a gasless option vs defaulting to gasless for all transfers of USDC on base-mainnet?

File: lib/coinbase/client/api/external_addresses_api.rb [hunk 2]
```diff
@@ -94,6 +94,91 @@ def get_external_address_balance_with_http_info(network_id, address_id, asset_id
       return data, status_code, headers
     end
 
+    # Get address balance history for asset
```
Comment (jazz-cb) [inline lib/coinbase/client/api/external_addresses_api.rb:97]: do you need both lines 97 and 98?

File: README.md (hunk 1 above)
Comment (alex-stone) [reply README.md:217]: That is something that we can enable by default if we'd like, but initially we were planning on not changing the default behavior.

We can roll this out as a default on the backend (but would be a breaking change for clients that do not yet support this signing of sponsored send messages)

File: lib/coinbase/address/wallet_address.rb [hunk 3]
```diff
@@ -39,8 +39,9 @@ def key=(key)
     # @param asset_id [Symbol] The ID of the Asset to send. For Ether, :eth, :gwei, and :wei are supported.
     # @param destination [Wallet | Address | String] The destination of the transfer. If a Wallet, sends to the Wallet's
     #  default address. If a String, interprets it as the address ID.
+    # @param gasless [Boolean] Whether the transfer should be gasless. Defaults to false.
```
Comment (jazz-cb) [inline lib/coinbase/address/wallet_address.rb:42]: should we explain what gasless is? "Whether gas fee for the transfer should be covered by Coinbase. This feature is only supported for USDC on Base-Mainnet. Defaults to false."

**Review by jazz-cb — APPROVED** [review]

File: lib/coinbase/address/wallet_address.rb (hunk 3 above)
Comment (alex-stone) [reply lib/coinbase/address/wallet_address.rb:42]: Yeah, I didn't want to make the code documentation be coupled to our backend limitations, namely which assets and networks we support.

E.g. we can add other network support or `:eurc` support without and client-side changes.

Definitely think it makes sense to have a section in our top-level docs indicating what specifically is supported though!

**Pull request merged**